lib/rom/sql/types/pg.rb

```diff
@@ -5,21 +5,49 @@ module ROM
   module SQL
     module Types
       module PG
-        Sequel.extension(:pg_json)
+        # UUID
+
+        UUID = Types::Strict::String.constrained(format: Regexp.new('^([0-9a-f]{8})-([0-9a-f]{4})-([0-9a-f]{4})-([0-9a-f]{2})([0-9a-f]{2})-([0-9a-f]{12})$'))
```

Comment:
You're right. I'm for removing that constraint for a few reasons:

  * It only supports UUID v4
  * It requires to be maintained
  * The original reason why I introduced it, was to control input and avoid malformed SQL statements (see _problem 1_)
  * We don't have a UUID type in Ruby. Indeed, `SecureRandom.uuid` returns an instance of `String`. We can keep `Types::PG::UUID` as a `Types::Strict::String` alias.

#### Problem 1

While using UUID as primary key, when the input isn't an UUID I get the following error:

```ruby
repository = SourceFileRepository.new
found = repository.find('foo')

# => 
#   Sequel::DatabaseError: PG::InvalidTextRepresentation: ERROR:  invalid input syntax for uuid: "foo"
      #   LINE 1: ...", "updated_at" FROM "source_files" WHERE ("id" = 'foo')...
```

I would expose a standardized behavior for missing records. When a table uses serial primary keys, if I do `repository.find('foo')` it returns `nil`. Tables with UUIDs as primary keys should behave the same.